app/main.py

```diff
@@ -1,3 +1,48 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: [float, int]) -> any:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {str(self.km)} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: [int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        else:
+            return Distance(self.km + other)
+
+    def __iadd__(self, other: [int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        else:
+            self.km += other
+            return self
+
+    def __mul__(self, other: [int, float, Distance]) -> Distance:
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: [int, float, Distance]) -> Distance:
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: [int, float, Distance]) -> bool:
+        return self.km < other
+
+    def __gt__(self, other: [int, float, Distance]) -> bool:
+        return self.km > other
+
+    def __eq__(self, other: [int, float, Distance]) -> bool:
+        return self.km == other
+
+    def __le__(self, other: [int, float, Distance]) -> bool:
+        return self.km <= other
+
+    def __ge__(self, other: [int, float, Distance]) -> bool:
+        return self.km >= other
```

Comment:
will it work with if other will be an instance of distance?
![image](https://user-images.githubusercontent.com/103201010/218129738-d3b32da3-5e3c-4e15-b1e3-2e4c14311e44.png)

Reply:
usualy no, but isn`t it an ability of Magic metods to do that (compare instances to a int(ex.)???

Reply:
so, i concider it will work!